### PR TITLE
enhancements: Propose a more formal yet lean enhancement process

### PR DESCRIPTION
**Pull request by fabiand**
**What this PR does / why we need it**:

Today it's a [very small group](https://github.com/kubevirt/community/blob/main/OWNERS#L7-L14)
which can approve design-proposals.
While certain individuals can approve design proposals, once they are merged
it is not clear who is owning them post-merge.

In this proposal we are proposing to change the process and shape them
around SIGs.
At the heart this proposal is requiring

1. designs to be sponsored by a SIG
2. to make a SIG responsible for the design process (reviews of design, code, documentation)
3. to make a SIG responsible for managing the design process (collaboration with other SIGs as needed)
4. to make a SIG owning the feature once it has been merged. The SIG is responsible for maintaing, fixing, running, everything it.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

Please see the "preview" at https://github.com/fabiand/enhancements

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [x] ~~Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)~~
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [x] ~~Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test~~
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [x] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by fabiand** [comment]
@aburdenthehand please review as well (you get this as your very own comment to invite you for review :) )

**Review by aburdenthehand — COMMENTED** [review]
Mostly suggestions but also a couple of questions about lifecycle

File: design-proposals/enhancements.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
+# Overview
+
+Rework the existing KubeVirt design proposal process into a more
+formal - still lean - Enhancement Porposal process.
```
[the remark this answers is not in the document]
Comment (lyarwood) [reply design-proposals/enhancements.md:4]: I'd also highlight this process will be aligned with SIGs.

File: design-proposals/enhancements.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+We move back to the community/design-proposals process.
+
+## Functional Testing Approach
+
+Require this process for KubeVirt v1.3+
+
+# Implementation Phases
+
+Beta for KubeVirt v1.3
+GA in KubeVirt v1.4
```
Comment (lyarwood) [inline design-proposals/enhancements.md:103]: I'd actually say Beta for 1.4 and GA for 1.5 given we only have ~49 days until 1.3 FF and hopefully most design proposals for the cycle have already been written.

File: design-proposals/enhancements.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+it is not clear who is owning them post-merge.
+
+In this proposal we are proposing to change the process and shape them
+around SIGs.
+At the heart this proposal is requiring
+
+1. designs to be sponsored by a SIG
+2. to make a SIG responsible for the design process (reviews of design, code, documentation)
+3. to make a SIG responsible for managing the design process (collaboration with other SIGs as needed)
+4. to make a SIG owning the feature once it has been merged. The SIG is responsible for maintaing, fixing, running, everything it.
```
[the remark this answers is not in the document]
Comment (lyarwood) [reply design-proposals/enhancements.md:20]: I'd suggest following k8s KEPs here so in such a case we always have an owning SIG that are responsible for the feature post merge and participating SIGs if the feature does cross SIG boundaries:

https://github.com/kubernetes/enhancements/blob/master/keps/sig-architecture/0000-kep-process/README.md#kep-metadata

File: design-proposals/enhancements.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+* As a VEP Author I want to know who I can work with in order to move
+  my proposal forward
+* As a SIG I want to have a say in what is getting pushed into my domain
+  in order to make sure that we are able to maintain it
+* As a SIG Approver I want to ensure that a design is sound before a
+  VEP Author is approaching an implementation.
+
+## Repos
+
+- https://github.com/kubevirt/enhancements
```
Comment (lyarwood) [inline design-proposals/enhancements.md:53]: Can I suggest that we create and seed this repo with examples to help folks understand the proposal better?

It can then also serve as a place to document follow ups or discussions we don't want to resolve prior to merging this document.

Happy to help with this if you don't have the bandwidth.

File: design-proposals/enhancements.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+- Ownership: SIGs own a freature from it's inception (design) all the way to it's end
+- Approvals: SIG approvers will be allowed to approve designs
+- Responsibilities: SIG Approvers are responsible for driving a design, and connecitng it to other SIGs as needed
+
+Process elements:
+
+- VEP Author creates a GitHub Issue for getting a unique identifier and starting the process
+- VEP Author creates a PR to propose the design targeting a specific SIG
+- SIG decides on an approver to sheppered the VEP
```
[the remark this answers is not in the document]
Comment (lyarwood) [reply design-proposals/enhancements.md:67]: Later on the document talks about mirroring the OWNERS files from core `kubevirt/kubevirt` so I would assume the core approval group would end up with this responsibility.

It might be cleaner to define a meta SIG like `sig-architecture` in k8s who manage the KEP process and keep PRs flowing there?

File: design-proposals/enhancements.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+And to distribute the work better.
+
+## Update/Rollback Compatibility
+
+We move back to the community/design-proposals process.
+
+## Functional Testing Approach
+
+Require this process for KubeVirt v1.3+
+
```
[the remark this answers is not in the document]
Comment (lyarwood) [reply design-proposals/enhancements.md:99]: +1 I wouldn't change the requirements here and would reword this as something like `strongly encourage` while also highlighting the benefits of getting a design reviewed alongside a PR etc.

**Review by lyarwood — COMMENTED** [review]
Added some comments. Overall I think the target of GA by 1.4 is too aggressive and we really need to create and seed the `kubevirt/enhancements` repo to help land this proposal. Happy to help with this!

File: design-proposals/enhancements.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (xpivarc) [reply design-proposals/enhancements.md:67]: Generally I think each SIG needs to have a representative in order to make sure someone from the SIG is reviewing PRs/proposals/issues, otherwise everyone can say the next one will have a look. I'll have a look what we have written down for now and will propose this.

>It might be cleaner to define a meta SIG like sig-architecture in k8s who manage the KEP process and keep PRs flowing there?

I think this could be beneficial for more reasons, common solutions on common problems across SIGs and so on. Therefore +1 from me

File: design-proposals/enhancements.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (xpivarc) [reply design-proposals/enhancements.md:99]: What about requiring a proposal if an new API is being proposed? The API is the biggest burden to maintain ,imho, so making sure we don't overextend it seems reasonable. Also, nowadays we have 3 months dev cycles while in the past we had only one month, therefore proposals are less likely to stall progress(if we do good work reviewing them...).

File: design-proposals/enhancements.md (hunk 2 above)
Comment (xpivarc) [reply design-proposals/enhancements.md:103]: +1

File: design-proposals/enhancements.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (lyarwood) [reply design-proposals/enhancements.md:99]: I wouldn't be against mandating this for enhancements touching the API for the 1.4 release cycle but I think our lives would be easier setting the bar a little lower until the process fully GAs.

**Comment by fabiand** [comment]
xlink #286

File: design-proposals/enhancements.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+## Repos
+
+- https://github.com/kubevirt/enhancements
+
+# Design
+
+Key elements:
+
+- Ownership: SIGs own a freature from it's inception (design) all the way to it's end
```
[the remark this answers is not in the document]
Comment (fabiand) [reply design-proposals/enhancements.md:59]: Very good point.

To me this should be defined in #251 and possibly #285.
Once they land, then this doc should reference to them - I'll add a note to the doc.

File: design-proposals/enhancements.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (fabiand) [reply design-proposals/enhancements.md:67]: What a word!

[the remark this answers is not in the document]
Comment (fabiand) [reply design-proposals/enhancements.md:67]: But `herd` makes these to be related to `shepherd`. This makes me wonder what the prefix `shep` donates.

File: design-proposals/enhancements.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (fabiand) [reply design-proposals/enhancements.md:20]: @davidvossel addressed. Now, I think we _must_ require one owning SIG, even if a fetaure is spread accross multiple. One SIG _must_ feel responsible to resolve issues or drive a feature forward, even if this requires extensive work with other SIGs.
In the end a feature has to have an owner, and it's better to have this hard discussoin through the inception of a feature - instead of waiting for a an urgency which could urge us to quickly find an owner.
Also: If we can not identify an owner for a feature, then we should ask ourselfs: Can we really take this feature if we are unable to own it?

(Just read Lee's comment now, but +1)

[the remark this answers is not in the document]
Comment (fabiand) [reply design-proposals/enhancements.md:20]: Reworded in order to express ownership to be exactly 1.

File: design-proposals/enhancements.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (fabiand) [reply design-proposals/enhancements.md:67]: The SIG owns a PR. If the SIG is not responsible, then we should define a group which a VEP author can reach out to get some support.

I do _not_ think that all PRs should run through i.e. something like sig-architecture.
A core principle of this proposal is to distribute the load.

At best a group like a throectical sig-architecture could act as a moderator between SIG and VEP author, but must not be directly involved with a VEP, unless the VEP is owned by this (arch) SIG.

@davidvossel would it be good enough for you to say: "In case of an unresponsive SIG, the VPE author can consult the KubeVirt Maintainers in order to resolve the situation."?

[the remark this answers is not in the document]
Comment (fabiand) [reply design-proposals/enhancements.md:67]: I've added:
> - KubeVirt Maintainers are responsible to support the owning SIG and VEP author in the case of conflicts and questions

File: design-proposals/enhancements.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (fabiand) [reply design-proposals/enhancements.md:99]: Let me slice it into two parts:

1. When does this proposal become effective (a new way to process features)
2. What features are - covered in #285

I merely meant to say:
For whatever feature is proposed accroding to current standards, from 1.3 on, let's use this newly proposed process.
This proposal does not change the requirement for when a design is required.

[the remark this answers is not in the document]
Comment (fabiand) [reply design-proposals/enhancements.md:99]: FWIW this should also be clear from the doc itself, as the proposal itself does not touch what a feature is.

File: design-proposals/enhancements.md (hunk 4 above)
Comment (fabiand) [reply design-proposals/enhancements.md:53]: Yes.

Please see https://github.com/fabiand/enhancements
This includes this proposal as a dry run, and I've got a PR to convert all existing - former - designs into this new format, specifically assigning primary sig owners for former designs.

File: design-proposals/enhancements.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (iholder101) [reply design-proposals/enhancements.md]: How about adding a label stating that a design proposal doesn't have an owning SIG yet? Something like `triage/needs-owning-sig`?

This way it would be more visible that the design proposal needs to be assigned + it would be easier querying such VEPs

[the remark this answers is not in the document]
Comment (fabiand) [reply design-proposals/enhancements.md]: Yes, that's a good idea. How good are you in prow automation coding, Itamar?

File: design-proposals/enhancements.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+* SIG Approver - A member of a SIG with approval permissions
+
+## User Stories
+
+* As a VEP Author I want to know who I can work with in order to move
+  my proposal forward
+* As a SIG I want to have a say in what is getting pushed into my domain
+  in order to make sure that we are able to maintain it
+* As a SIG Approver I want to ensure that a design is sound before a
+  VEP Author is approaching an implementation.
```
Comment (iholder101) [inline design-proposals/enhancements.md:50]: Maybe add:
* As a Kubevirt developer, I want to have a single source-of-truth w.r.t. the current status, design API of a feature.

(see my comment below regarding this topic)

File: design-proposals/enhancements.md [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+- `OWNER_ALIASES` will be mirrored from kubevirt/kubevirt in order to have the same SIGs in the EP repository
+- Approvals and ownership is defined with `OWNERS` files in the `veps/sig-*` directories, tying into the general prow approval and merge flow
+- GitHub Issues will be used to create unique identifiers
+
+## API Examples
+
+None.
+
+## Scalability
+
```
Comment (iholder101) [inline design-proposals/enhancements.md:88]: nit: redundant newline

**Review by iholder101 — COMMENTED** [review]
Thanks @fabiand!
Overall looks great!

File: design-proposals/enhancements.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (fabiand) [reply design-proposals/enhancements.md]: Thanks :)

FWIW - I think that we need approvers and "potential chairs" to really look into automation to simply help the process. the appovers and pr authors.

File: design-proposals/enhancements.md [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+This has a few effects - see the following Goals section.
+
+## Goals
+
+1. The design process now has a mechanism to distribute designs among SIGs
+2. SIG approvers are empowered to approve designs, increase the approvers pool
+3. The ownership of an implemented feature becomes clear
+4. Ensure that designs converge (accept, reject)
+5. Formalize this process as an Virtualization Enhancement Proposal (VEP) process
```
[the remark this answers is not in the document]
Comment (fabiand) [reply design-proposals/enhancements.md:31]: Yeah, I thought about this as well.

My worry would be the confusion between kube and kubevirt KEPs.

File: design-proposals/enhancements.md [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+Process elements:
+
+- VEP Author creates a GitHub Issue for getting a unique identifier and starting the process
+- VEP Author creates a PR to propose the design targeting a specific SIG
+- SIG decides on an approver to shepherd the VEP
+- SIG collaborates with other SIGs to ensure its thoroughly reviewed
+- SIG approves or rejects VEP
+- Other SIG approvers can veto a proposal
+- SIG owns future maintenance of the implementation
+- KubeVirt Maintainers are responsible to support the owning SIG and VEP author in the case of conflicts and questions
```
[the remark this answers is not in the document]
Comment (fabiand) [reply design-proposals/enhancements.md:74]: @iholder101 very well. What would you add to make it clear that it is a living doucment?

> BTW, in k8s the tracking issue remains open until the feature GAs.

This is part of this proposal as well. If this is not clear, then it should be clarified.

[the remark this answers is not in the document]
Comment (fabiand) [reply design-proposals/enhancements.md:74]: Ah-a. I see your comment below.

File: design-proposals/enhancements.md (hunk 8 above)
Comment (fabiand) [reply design-proposals/enhancements.md:50]: Perect, agreed, added.

File: design-proposals/enhancements.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (iholder101) [reply design-proposals/enhancements.md:74]: WDYT about the following? I've highlighted new bullets

- VEP Author creates a GitHub Issue for getting a unique identifier and starting the process
- VEP Author creates a PR to propose the design targeting a specific SIG
  - **VEP Author is expected to list requirements and goals for Alpha**
  - **Preferably, VEP Author also vaguely lists requirements for Beta/GA.**
- SIG decides on an approver to shepherd the VEP
- SIG collaborates with other SIGs to ensure its thoroughly reviewed
- SIG approves or rejects VEP
- Other SIG approvers can veto a proposal
- SIG owns future maintenance of the implementation
- **VEP author should make sure that the VEP is the source of truth w.r.t. the VEP's status (design / implementations / goals / etc)**
  - **VEP author should create more PRs to update the VEP whenever it is not reflecting the updated VEP's status**
  - **VEP author updates the requirements for Beta/GA as they become clearer**
  - **VEP author updates implementation phases for Beta/GA as they become clearer**
- KubeVirt Maintainers are responsible to support the owning SIG and VEP author in the case of conflicts and questions

File: design-proposals/enhancements.md [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+2. to make one SIG responsible for the design process (reviews of design, code, documentation)
+3. to make one SIG responsible for managing the design process (collaboration with other SIGs as needed)
+4. to make one SIG to own the feature once it has been merged. The SIG is responsible for maintaining, fixing, running, _everything-it_.
+
+This has a few effects - see the following Goals section.
+
+## Goals
+
+1. The design process now has a mechanism to distribute designs among SIGs
+2. SIG approvers are empowered to approve designs, increase the approvers pool
```
Comment (jean-edouard) [inline design-proposals/enhancements.md:28]: Unclear. SIG approvers can increase the approvers pool?
Or, the fact that SIG approvers can approve designs increases the approvers pool? In which case:
```suggestion
2. SIG approvers are empowered to approve designs, increasing the approvers pool
```
But does it really? Technically, most sigs have less approvers than kubevirt/community does today...

File: design-proposals/enhancements.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (fabiand) [reply design-proposals/enhancements.md:76]: I think this is going in the right direction, I'll update the doc

File: design-proposals/enhancements.md [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+
+In this proposal we are proposing to change the process and shape them
+around SIGs.
+
+At its core, this proposal requires:
+
+1. designs to be sponsored by a SIG
+2. to make one SIG responsible for the design process (reviews of design, code, documentation)
+3. to make one SIG responsible for managing the design process (collaboration with other SIGs as needed)
+4. to make one SIG to own the feature once it has been merged. The SIG is responsible for maintaining, fixing, running, _everything-it_.
```
[the remark this answers is not in the document]
Comment (fabiand) [reply design-proposals/enhancements.md:21]: Yes, yes, it was more like a small (failed) word game:

maintain it
fix it
run it
everything it ....

File: design-proposals/enhancements.md (hunk 12 above)
Comment (fabiand) [reply design-proposals/enhancements.md:28]: The second. Because we have sig approvers, we implicitly increase the pool.

Yes, sigs have less approvers then root, but: the sum of sig approvers should venetually be larger than the amount of root approvers.

**Comment by fabiand** [comment]
@jean-edouard @iholder101 @lyarwood @0xFelix @vladikr @xpivarc  i'd appreciate your opinion to start moving fwd with this.

File: design-proposals/enhancements.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (alaypatel07) [reply design-proposals/enhancements.md]: @lyarwood @fabiand @xpivarc: can we bring this discussion topic to sig-api call on June 4th? I think we could potentially find a way to make progress on making design requirements for changes in API in a gradual way the does not overwhelm participants yet makes iterative progress in coming releases

File: design-proposals/enhancements.md [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+
+## API Examples
+
+None.
+
+## Scalability
+
+Instead of relying on a small approvers pool, now the process starts
+with routing VEPs in the beginning of their life-time to SIGs.
+This is expected to increase the time-to-merge-or-reject for VEPs.
```
Comment (0xFelix) [inline design-proposals/enhancements.md:93]: Does this really increase the time? I would think spreading the load across a larger pool of possible approvers should speed up the process?

File: design-proposals/enhancements.md (hunk 13 above)
Comment (0xFelix) [inline design-proposals/enhancements.md:21]: Shouldn't this process encourage a VEP author to become member of said SIG? Currently it sounds a bit like as a VEP author I can forget about the feature once it is merged and the SIG has to carry the burden of maintaining it.

File: design-proposals/enhancements.md [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+it is not clear who is owning them post-merge.
+
+In this proposal we are proposing to change the process and shape them
+around SIGs.
+
+At its core, this proposal requires:
+
+1. designs to be sponsored by a SIG
+2. to make one SIG responsible for the design process (reviews of design, code, documentation)
+3. to make one SIG responsible for managing the design process (collaboration with other SIGs as needed)
```
Comment (0xFelix) [inline design-proposals/enhancements.md:20]: To me these two items essentially are the same, maybe merge them?

File: design-proposals/enhancements.md [hunk 16]
```diff
@@ -0,0 +1,109 @@
...
+
+* VEP Author - The person writing a design/enhancement proposal
+* SIG - A formal KubeVirt SIG
+* SIG Approver - A member of a SIG with approval permissions
+
+## User Stories
+
+* As a VEP Author I want to know who I can work with in order to move
+  my proposal forward
+* As a SIG I want to have a say in what is getting pushed into my domain
```
Comment (0xFelix) [inline design-proposals/enhancements.md:47]: Will this hurt adoption of new features in the long run if a SIG decides there is no capacity to maintain everything? I'm thinking of a constantly increasing maintenance load on the SIG while the count of members may not increase.

File: design-proposals/enhancements.md [hunk 17]
```diff
@@ -0,0 +1,109 @@
...
+Key elements:
+
+- Ownership: SIGs own a _feature_ (which includes the process, but also
+  the resulting code) from it's inception (design) all the way (fixes, maintenance) to it's end (removal)[^1]
+- Approvals: SIG approvers will be allowed to approve designs
+- Responsibilities: SIG Approvers are responsible for driving a design, and connecting it to other SIGs as needed
+
+Process elements:
+
+- VEP Author creates a GitHub Issue for getting a unique identifier and starting the process
```
Comment (0xFelix) [inline design-proposals/enhancements.md:69]: nit: Add link to the repo?

File: design-proposals/enhancements.md (hunk 13 above)
Comment (iholder101) [reply design-proposals/enhancements.md:21]: > Currently it sounds a bit like as a VEP author I can forget about the feature once it is merged and the SIG has to carry the burden of maintaining it.

Do you mean "once it GAs"?
If you'd forget about the feature before it GAs, it will eventually be dropped.

Forgetting about a feature after developing it to the point it GAs sounds fair to me TBH.

> I would add that the original author is encouraged but not required to become a member of the SIG to help with on-going maintenance of the feature they are introducing.

I'm not against encouraging it, but not sure it's necessary TBH. These scenarios are completely valid IMO:
1. Someone develops a feature under a certain SIG but is interested only in the feature's narrow scope, not the entire SIG. For example, someone can develop a network feature under sig-network, but care about this specific feature and not about SIG network generally.
2. Someone can develop a feature, own and maintain it for a while, then stop maintaining it (due to switching his position, becoming interested in other stuff, etc). IOW, we should expect that people will move around. In any case the SIG always owns all of its features while people can stop maintaining them.

File: design-proposals/enhancements.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (iholder101) [reply design-proposals/enhancements.md:31]: @fabiand yeah this is a valid concern imo

File: design-proposals/enhancements.md (hunk 16 above)
Comment (iholder101) [reply design-proposals/enhancements.md:47]: @lyarwood +1.

It is better to understand that we don't have the capacity to maintain a feature. Now we can't even figure that out because there is no clear definition of ownership.

File: design-proposals/enhancements.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (iholder101) [reply design-proposals/enhancements.md:76]: This is still relevant I think

**Comment by iholder101** [comment]
@fabiand @0xFelix @jean-edouard @lyarwood @davidvossel 

I think this proposal is very important and would serve as a perfect alternative to the current "design proposal" which is not owned by any SIG, doesn't serve as a live-document until the feature GAs, and so on.

What's left in order to get this in? Does anyone have any objections?

**Comment by jean-edouard** [comment]
> @fabiand @0xFelix @jean-edouard @lyarwood @davidvossel
> 
> I think this proposal is very important and would serve as a perfect alternative to the current "design proposal" which is not owned by any SIG, doesn't serve as a live-document until the feature GAs, and so on.
> 
> What's left in order to get this in? Does anyone have any objections?

Honestly I'm a bit concerned about this.
SIGs created big silos within the project/community/CI.
To me, this proposal deepens those silos by encouraging members of a given SIG to both submit and approve design proposals. That looks a lot like a conflict of interest!
I would almost argue for the opposite, i.e. design proposals must be approved by someone who is *not* a member of that SIG...

**Comment by iholder101** [comment]
> > @fabiand @0xFelix @jean-edouard @lyarwood @davidvossel
> > I think this proposal is very important and would serve as a perfect alternative to the current "design proposal" which is not owned by any SIG, doesn't serve as a live-document until the feature GAs, and so on.
> > What's left in order to get this in? Does anyone have any objections?
> 
> Honestly I'm a bit concerned about this. SIGs created big silos within the project/community/CI. To me, this proposal deepens those silos by encouraging members of a given SIG to both submit and approve design proposals. That looks a lot like a conflict of interest! I would almost argue for the opposite, i.e. design proposals must be approved by someone who is _not_ a member of that SIG...

Man, that's an interesting thought! :)

As I see it, this proposal has two main goals:
1. Requiring an issue that would serve as a source of truth regarding a VEP's progress (similar to k8s KEPs).
2. Easing the process of design-proposals/VEPs getting merged.

We aim to address (2) by introducing SIGs and giving them ownership and responsibility of the KEPs in their domain. This way SIGs can approve proposals without waiting for many "uber-approvers" to chime in the discussion, and have the responsibility to respond in a reasonable time frame.

> That looks a lot like a conflict of interest!

I understand your point. However, I'm not sure it will hurt us in practice for a few reasons.
1. A SIG does not only submit and approve proposals, but also maintains them. Rushing into approving a bad proposal is basically shooting yourself in the foot.
2. This proposal [writes](https://github.com/kubevirt/community/pull/288/files#diff-33df9c097ff4b3126c617cfff31dbe1fb6f738d8dd9124341a387f2370e815cbR72): `SIG collaborates with other SIGs to ensure its thoroughly reviewed`. IOW It's the SIG's responsibility to ensure which other SIGs are related to the proposal by actively reaching out to them and getting their ACK. If any one of the other SIGs thinks it relates to them as well, I agree that we should require an approval/LGTM from their side in order to accept the proposal.
3. We should anyways encourage other SIGs (and all community members!) to take part in the discussions about the different proposals. The fact that a SIG owns a VEP doesn't mean that others' opinion do not count.
4. Requiring that every proposal should be reviewed by all KEPs kind of misses the point of addressing (2). We would again get to a point where it would be pretty hard to get a reviewer to look at your proposal.

Let me know what you think

**Comment by iholder101** [comment]
@jean-edouard ping :)

@fabiand WDYT about splitting this proposal into two - one for using a tracking issue with graduation requirements etc, and another one for the SIG ownership part? I support them both, but perhaps it'll be easier to merge this way.

**Review by enp0s3 — CHANGES_REQUESTED** [review]
@fabiand Thank you for the PR, I think the implementation phases section needs to be updated

File: design-proposals/enhancements.md [hunk 18]
```diff
@@ -0,0 +1,109 @@
...
+
+## Functional Testing Approach
+
+Require this process for KubeVirt v1.4 and onwards.
+
+# Implementation Phases
+
+Beta for KubeVirt v1.4
+GA in KubeVirt v1.5
+
```
Comment (enp0s3) [inline design-proposals/enhancements.md:108]: @fabiand Would you like some more help with the PR? perhaps adding collaborators?

**Comment by aburdenthehand** [comment]
@fabiand Now that #362 is merged, should we close this?

**Comment by aburdenthehand** [comment]
/close
Now that #362 is merged and the enhancements repo is under trial for v1.5.
